input-output-hk/ouroboros-network
# Bump versions

- Fix Praos obsolete node check
- cardano-tools: new project (squashed)
- cardano-tools: (styling) rearrange import groups
- cardano-tools: (styling) disappear `..` wildcards from project code
- cardano-tools: provide command line options and usage in comments
- cardano-tools: condition locally defined
- cardano-tools: styling; more explicit test case
- cardano-tools: documentation; test-case fix
- cardano-tools: minor style fixes
- db-synthesizer: enhance with append mode
- Bug fix: base `maxMajorProtVer` off of Babbage
- version bump to 0.1.0.1
- improve/add documentation
- GR-FIXME: comments and fixme's from Galois Review
- refactors
- refactor P2P api
- Address review comments.
- Address comments regarding ControlChannel
- Refused renaming of ProtocolTemperature
- Addressed review comments.
- Renamed Bundle as TemperatureBundle
- Race threads via `foldl1`.
- Addressed review comments in diffusion
- Move P2P snocket utils to Ouroboros.Network.Diffusion.Utils module
- Removed always true assertion
- Renamed splitPeers and partitionPeer
- Fixed TOC of Ouroboros.Network.PeerStateActions haddocks
- Removed PeerState date type
- Change type signature of withPeerSelectionActions
- Renamed TokProtocolTemperature as SingProtocolTemperature
- showConnectionEvents: removed unnecessary brackets
- JobPool: renamed collect to waitForJob
- JobPool: improve haddocks, clean the code
- Peer list lifetime constants, improved haddocks.
- Added server_CONNABORTED_DELAY constant.
- Move a comment to the right place
- Configurable inbound-governor idle timeout
- Simplify DebugPeerSelection type using existential type variable
- Added initiator mini-protocols result type
- ExitPolicy - compute reconnect delay
- Use ExitPolicy to compute reconnection delay
- Pass reconnect error delay through PeerSelectionPolicy
- Updated ouroboros-network:test component
- Updated ouroboros-consensus
- Updated ouroboros-consensus-test
- Fixed code layout
- diffusion-test: pass seed through SimArgs
- Simulation.Node: renamed qualified import of NodeKernel
- Fixed show instances of some types
- Organise exports for the testnet simulation
- Fixed prop_diffusion_target_active_local_above
- inbound-governor: code style
- outbound-governor: fixed haddock comments
- peer-metrics: use a single TVar for upstreameness and fetchyness metrics
- peer-metrics: added export list, improved haddocks.
- peer-metrics: removed getHeaderMetrics and getFetchedMetrics
- peer-metrics: change IntPSQ qualified name
- churn: added configurable churn interval
- peer-metrics: added PeerMetricsConfiguration
- peer-metrics: use averages to instantiate the peer metrics
- TraceLabelPeer: added Bifunctor instance
- data-script: Added maybeStepScript and Delay
- peer-metrics: resolve ties of hot demotion with slot number
- peer-metrics: added tests
- peer-selection: Improved haddock of 'TracePromoteWarmLocalPeers'
- peer-metrics: code style
- Added FetchClientRegistry to NodeKernel Used ClientRegistry bracket functions
- Added Hashable Orphan instances
- Added header body-split block-fetch to diffusion
- Added HotDiffusionScript
- Remove time limit on events
- stylish
- Disables diffusion tests in windows
- Fixes #3935
- Reinitialise dns library
- asyncResolverResource: use resolverResource on Windows
- dns-actions: fixed a typo
- dns-actions: provide getResolver function
- dns-actions: make internal loops tail recursive
- Fixed stylish-haskell parsing error
- Rebuild storage-test component
- Updated network dependency
- snocket: various stylistic changes
- snocket: do not configure sockets in bind method
- connection-manager: add a method to configure connection sockets
- socket: added methods to configure sockets
- subscription: exception handling
- subscription: configure sockets
- subscription: enhance test output
- rethrow-policies: fixed typos
- connection-handler: rethrow exceptions using ExceptionInHandler
- ouroboros-newtork-framework: changed ExceptionInHandler
- ouroboros-consensus: added top level error tracer
- diffusion: DiffusionTracer
- Updated CHANGELOG files
- Use newer version of io-sim
- connection-manager: reworded some comments
- peer-state-actions: make monitorPeerSelection non-blocking
- peer-state-actions: use updateUnlessCold
- peer-state-actions: use epErrorDelay
- exit-policy: derive semigroup via Max
- exit-policy: derive Fractional instance for ReconnectDelay
- peer-selection: trace reconnect delay with 'TraceDemoteAsynchronous'
- peer-selection: refactor monitoring connections
- peer-selection: added localRoots to PeerSelectionCounters
- Updated docs/interface-CHANGELOG.md file
- peer-selection: throw an exception when monitorPeerSelection blocks
- Refactored DiffusionScript
- Stylish
- Update ouroboros-network/test/Test/Ouroboros/Network/Testnet/Simulation/Node.hs
- connection-manager: haddocks, code style & typos
- net-sim: exported WithAddr
- net-sim: protect from establishing connection twice
- net-sim: code cleanup
- diffusion-sim: added prop_diffusion_target_active_local_above
- diffusion-sim: reproduce peer-state-action bug
- diffusion-sim: set keep-alive and ping-pong intervals
- diffusion-sim: refactor how a tracer is used
- diffusion-sim: fix ping pong time limits
- diffusion-sim: cleanup tracers
- diffusion-sim: added debugTracer
- peer-state-actions: export traces
- signals: do not ignore tail in keyedTimeout
- signals: added fromEventsWith
- diffusion-sim: added async demotion test
- diffusion-sim: refactor test cases
- diffusion-test: use more relaxed imports
- Fixed a few typos and code layout
- DNSActions: refactor code
- dns-actions: make sure TTLs are strictly greater than 0
- Support NodeToNodeV_10
- Fix chocolatey CI error
- Windows Github Actions now use MSYS2
- Fix GH Actions Windows CI caching
- Updated versions of github actions
- github-actions: use `ubuntu-latest`
- Fix GH Actions Windows CI: switch from pkgconf to pkg-config
- github-actions: update cabal before checking out the repository
- Run GitHub Actions on pull requests
- Update io-sim & its bounds
- Update typed-protocols & its bounds
- peer-selection: added TraceDemoteLocalAsynchronous trace
- Fixed stylish-haskell
- github-actions: use $GITHUB_OUTPUT variable
- github-actions: fixed a installation failure on Windows
- Use newer version of io-classes, io-sim and strict-stm
- Updated typed-protocols
- EarlyExit - removed unused pragma
- Bump versions of packages
- Updated dependencies to match `cardano-node@release/1.35`
- crypto components: add secp256k1 to pkgconfig
- Bump version of ouroboros-network-framework-testing
- keep-alive: pass `ControlMessageSTM` to keep alive application
- Update SendFetchRequest trace
- Fix windows GHA (#4159)
- Bump versions
